lib/daru/io/exporters/excel.rb

```diff
@@ -0,0 +1,47 @@
+require 'daru/io/exporters/linkages/excel'
+
+module Daru
+  module IO
+    module Exporters
+      class ExcelHelper
+        # @note
+        #
+        # The +format+ variable used in this method, has to be given
+        # as options by the user via the +opts+ hash input.
+        #
+        # Signed off by @athityakumar on 03/06/2017 at 7:00PM
+        def prepare
+          @book  = Spreadsheet::Workbook.new
+          sheet  = @book.create_worksheet
+
+          format = Spreadsheet::Format.new color: :blue, weight: :bold
+
+          sheet.row(0).concat(@dataframe.vectors.to_a.map(&:to_s)) # Unfreeze strings
+          sheet.row(0).default_format = format
+          i = 1
+          @dataframe.each_row do |row|
+            sheet.row(i).concat(row.to_a)
+            i += 1
+          end
+        end
+
+        def write
+          prepare
+          @book.write(@path)
+        end
+      end
+
+      class Excel < ExcelHelper
```

Comment:
The `ExcelHelper` is intended to be used to populate those methods that would have been private methods in `Excel` class. As `daru-views` might need to access these private class methods (say, for the 'Download' feature - it has to prepare the data and not write the data), I've made the `ExcelHelper` class. If this is not intuitive, we can switch back to having one class with private methods.